File: 2021/UICPC DIV1/UICPC Round #12 (Div 1)/solutions/Problem A - Battleship/battleship.cpp
#include <bits/stdc++.h>
using namespace std;

// Battleship

int n, m, q;
char p[3][33][33];

int main(){
    int t;
    cin >> t;
    while(t--){
        cin >> m >> n >> q;
        int s1 = 0, s2 = 0;
        for(int i = n-1; i >= 0; i--)
            for(int j = 0; j < m; j++){
                cin >> p[0][i][j];
                if(p[0][i][j] == '#')
                    s1++;
            }
        for(int i = n-1; i >= 0; i--)
            for(int j = 0; j < m; j++){
                cin >> p[1][i][j];
                if(p[1][i][j] == '#')
                    s2++;
            }
        bool turn = true, end = false;
        while(q--){
            int x, y;
            cin >> y >> x;
            if(end)
                continue;
            if(p[turn][x][y] == '#'){
                p[turn][x][y] = '_';
                if(turn){
                    s2--;
                    if(!s2) turn = !turn;
                }
                else{
                    s1--;
                    if(!s1) end = true;
                }
            }
            else turn = !turn;
        }
        if(s1 && !s2)
            cout << "player one wins" << endl;
        else if(!s1 && s2)
            cout << "player two wins" << endl;
        else 
            cout << "draw" << endl;
    }
}

File: 2021/UICPC DIV1/UICPC Round #12 (Div 1)/solutions/Problem A - Battleship/battleship.py
t = int(input())
for i in range(t):
    w, h, n = map(int, input().split())
    pg = [[], []]
    sh = [0, 0]
    for j in range(h):
        pg[0].append(input())
        sh[0] += pg[0][j].count('#')
    pg[0].reverse()
    for j in range(h):
        pg[1].append(input())
        sh[1] += pg[1][j].count('#')
    pg[1].reverse()
    turn = 1
    fl = False
    for k in range(n):
        c, r = map(int, input().split())
        if fl and turn:
            continue
        if pg[turn][r][c] == '#':
            pg[turn][r] = pg[turn][r][0:c] + '_' + pg[turn][r][c+1:]
            sh[turn] -= 1
            if not sh[turn]:
                turn = 0 if turn else 1
                fl = True
        else:
            turn = 0 if turn else 1
    if sh[0] == sh[1] == 0 or (sh[0] != 0 and sh[1] != 0):
        print('draw')
    else:
        print('player two wins' if not sh[0] else 'player one wins')

